### PR TITLE
[ML] Remove redundant autodetect options

### PR DESCRIPTION
**Pull request by edsavage**
Remove the now redundant `autodetect` command line options and
exit with error message on seeing unknown command line arguments

### CODE REVIEW

File: bin/autodetect/Main.cc [hunk 1]
```diff
@@ -152,6 +151,12 @@ int main(int argc, char** argv) {
     // statically links its own version library.
     LOG_DEBUG(<< ml::ver::CBuildInfo::fullInfo());
 
+    if (unknownTokens.size()) {
+        LOG_ERROR(<< "Unkown command line options: "
+                  << ml::core::CContainerPrinter::print(unknownTokens));
+        return EXIT_FAILURE;
+    }
```
Comment (droberts195) [inline bin/autodetect/Main.cc:158]: You won't need this check if you configure Boost Program Options to ban unknown arguments.

File: bin/autodetect/CCmdLineParser.cc [hunk 2]
```diff
@@ -69,24 +68,14 @@ bool CCmdLineParser::parse(int argc,
                     "Optional field config file")
             ("modelplotconfig", boost::program_options::value<std::string>(),
                     "Optional model plot config file")
```
Comment (droberts195) [inline bin/autodetect/CCmdLineParser.cc:70]: Can `modelplotconfig`, `fieldconfig` and `limitconfig` also be removed now?

Also (nit) I just noticed the help text for `filtersconfig` and `eventsconfig` is not indented the same amount as all the other help text.

File: bin/autodetect/CCmdLineParser.cc [hunk 3]
```diff
@@ -107,19 +96,13 @@ bool CCmdLineParser::parse(int argc,
...
-            ("maxQuantileInterval", boost::program_options::value<core_t::TTime>(),
-                    "Optional interval at which to periodically output quantiles if they have not been output due to an anomaly - if not specified then quantiles will only be output following a big anomaly")
+                    "Optional number of buckets after which to periodically persist model state.")
             ("maxAnomalyRecords", boost::program_options::value<std::size_t>(),
                     "The maximum number of records to be outputted for each bucket. Defaults to 100, a value 0 removes the limit.")
             ("memoryUsage",
                     "Log the model memory usage at the end of the job")
-            ("multivariateByFields",
-                    "Optional flag to enable multi-variate analysis of correlated by fields")
             ("stopCategorizationOnWarnStatus",
```
Comment (droberts195) [inline bin/autodetect/CCmdLineParser.cc:106]: This one can come from the job config too now, and shouldn't be supported as a command line option.

File: bin/autodetect/CCmdLineParser.cc [hunk 4]
```diff
@@ -48,7 +47,7 @@ bool CCmdLineParser::parse(int argc,
                            std::size_t& maxAnomalyRecords,
                            bool& memoryUsage,
                            bool& stopCategorizationOnWarnStatus,
```
Comment (droberts195) [inline bin/autodetect/CCmdLineParser.cc:49]: Shouldn't be needed any more.

File: bin/autodetect/CCmdLineParser.cc [hunk 5]
```diff
@@ -107,19 +96,13 @@ bool CCmdLineParser::parse(int argc,
...
                     "Optional file to persist state to - not present means no state persistence")
             ("persistIsPipe", "Specified persist file is a named pipe")
-            ("persistInterval", boost::program_options::value<core_t::TTime>(),
-                    "Optional time interval at which to periodically persist model state (Mutually exclusive with bucketPersistInterval)")
             ("persistInForeground", "Persistence occurs in the foreground. Defaults to background persistence.")
             ("bucketPersistInterval", boost::program_options::value<std::size_t>(),
-                    "Optional number of buckets after which to periodically persist model state (Mutually exclusive with persistInterval)")
-            ("maxQuantileInterval", boost::program_options::value<core_t::TTime>(),
-                    "Optional interval at which to periodically output quantiles if they have not been output due to an anomaly - if not specified then quantiles will only be output following a big anomaly")
+                    "Optional number of buckets after which to periodically persist model state.")
```
Comment (droberts195) [inline bin/autodetect/CCmdLineParser.cc:101]: Is it now that if `bucketPersistInterval` is specified on the command line then it takes precedence over the persist interval in the config?

File: bin/autodetect/CCmdLineParser.h [hunk 6]
```diff
@@ -60,7 +59,7 @@ class CCmdLineParser {
                       std::size_t& maxAnomalyRecords,
                       bool& memoryUsage,
                       bool& stopCategorizationOnWarnStatus,
-                      TStrVec& clauseTokens);
+                      TStrVec& unknownTokens);
```
Comment (droberts195) [inline bin/autodetect/CCmdLineParser.h:62]: This argument can be removed.

File: bin/autodetect/CCmdLineParser.cc (hunk 5 above)
Comment (edsavage) [reply bin/autodetect/CCmdLineParser.cc:101]: Yes, we don't expect the `bucketPersistInterval` to ever be used in production code (it was introduced to support the `model_extractor` project). So if present it should take precedence over the background persist interval.

File: bin/autodetect/Main.cc [hunk 7]
```diff
@@ -110,16 +109,15 @@ int main(int argc, char** argv) {
     bool isPersistInForeground{false};
     std::size_t maxAnomalyRecords{100};
     bool memoryUsage{false};
-    bool stopCategorizationOnWarnStatus{false};
-    TStrVec clauseTokens;
+    TStrVec unknownTokens;
```
Comment (droberts195) [inline bin/autodetect/Main.cc:112]: I think this variable is now redundant.

**Review by droberts195 — APPROVED** [review]
LGTM if you could just delete 1 more redundant line

**Pull request merged**